runtime/parachains/src/reward_points.rs

```diff
@@ -22,42 +22,71 @@
 //! for the time being, although we will build schemes to do so in the future.
 
 use crate::{session_info, shared};
-use frame_support::traits::ValidatorSet;
-use primitives::v2::ValidatorIndex;
+use frame_support::traits::{Defensive, ValidatorSet};
+use primitives::v2::{SessionIndex, ValidatorIndex};
+use sp_std::collections::btree_set::BTreeSet;
 
 /// The amount of era points given by backing a candidate that is included.
 pub const BACKING_POINTS: u32 = 20;
+/// The amount of era points given by dispute voting on a candidate.
+pub const DISPUTE_STATEMENT_POINTS: u32 = 15;
 
 /// Rewards validators for participating in parachains with era points in pallet-staking.
 pub struct RewardValidatorsWithEraPoints<C>(sp_std::marker::PhantomData<C>);
 
-impl<C> crate::inclusion::RewardValidators for RewardValidatorsWithEraPoints<C>
+impl<C> RewardValidatorsWithEraPoints<C>
 where
-	C: pallet_staking::Config + shared::Config + session_info::Config,
+	C: pallet_staking::Config + session_info::Config,
 	C::ValidatorSet: ValidatorSet<C::AccountId, ValidatorId = C::AccountId>,
 {
-	fn reward_backing(indices: impl IntoIterator<Item = ValidatorIndex>) {
-		// Fetch the validators from the _session_ because sessions are offset from eras
-		// and we are rewarding for behavior in current session.
-		let session_index = shared::Pallet::<C>::session_index();
+	/// Reward validators in session with points, but only if they are in the active set.
+	fn reward_only_active(
```

Comment:
Maybe DQ: Shouldn't we be checking also they participated in parachain consensus ? The active set is much larger than the `max_validators` setting (200 on Kusama/Polkadot).

Reply:
`account_keys` is restricted to paravalidators, so it's filtered in `filter_map(|i| validators.get(i.0 as usize).cloned())`. 
The reason to also filter those in the active set is because staking pallet can only reward current era (or even session) active validators AFAIK.